pkg/globalnet/main.go

```diff
@@ -91,24 +96,18 @@ func main() {
 		os.Exit(1)
 	}
 
-	gatewayMonitor, err := ipam.NewGatewayMonitor(&ipamSpec, cfg, stopCh)
+	gatewayMonitor, err := ipam.NewGatewayMonitor(&ipamSpec, submarinerClient, clientSet)
 	if err != nil {
 		klog.Fatalf("Error creating gatewayMonitor: %s", err.Error())
 	}
 
-	var wg sync.WaitGroup
-
-	wg.Add(1)
-
-	go func() {
-		defer wg.Done()
+	if err = gatewayMonitor.Start(stopCh); err != nil {
+		klog.Fatalf("Error running gatewayMonitor: %s", err.Error())
+	}
 
-		if err = gatewayMonitor.Run(stopCh); err != nil {
-			klog.Fatalf("Error running gatewayMonitor: %s", err.Error())
-		}
-	}()
+	<-stopCh
+	gatewayMonitor.Stop()
```

Comment:
Okay, along with the cleanup when the process itself dies, we also want the cleanup (of certain iptable chains) even when there is a Gateway Migration.